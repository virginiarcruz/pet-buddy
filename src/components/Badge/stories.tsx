import { Story, Meta } from '@storybook/react/types-6-0'
import { Badge } from '.'

export default {
  title: 'Badge',
  component: Badge
} as Meta

export const Default: Story = (args) => <Badge labelBadge="28" {...args} />
