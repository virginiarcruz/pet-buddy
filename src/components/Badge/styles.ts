import styled, { css } from 'styled-components'

export const BadgeContainer = styled.span`
  ${({ theme: { border, color, font } }) => css`
    border-radius: ${border.radius.pill};
    background-color: ${color.orange};
    color: ${color.white};
    font-weight: ${font.bold};
    font-size: ${font.size.small};
    padding: 4px 8px;
    width: max-content;
    height: 25px;
    line-height: 17px;
  `}
`
