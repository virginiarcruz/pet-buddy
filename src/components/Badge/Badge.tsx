import { BadgeContainer } from './styles'

export type BadgeProps = {
  labelBadge?: string
}
export const Badge = ({ labelBadge }: BadgeProps) => (
  <BadgeContainer>{labelBadge}</BadgeContainer>
)
