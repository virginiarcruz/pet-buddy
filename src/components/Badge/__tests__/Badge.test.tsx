import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import { Badge } from '../Badge'

describe('<Badge />', () => {
  it('should render the badge', () => {
    const { container } = renderWithTheme(<Badge labelBadge="badge" />)

    expect(container).toMatchSnapshot()

    expect(screen.getByText('badge')).toBeInTheDocument()
    expect(screen.getByText('badge')).toHaveStyle({
      'font-size': '14px',
      'background-color': '#F47500'
    })
  })
})
