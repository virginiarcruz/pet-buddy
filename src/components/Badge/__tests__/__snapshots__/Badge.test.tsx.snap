// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> should render the badge 1`] = `
.c0 {
  border-radius: 100px;
  background-color: #F47500;
  color: #FFFFFF;
  font-weight: 600;
  font-size: 14px;
  padding: 4px 8px;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  line-height: 17px;
}

<div>
  <span
    class="c0"
  >
    badge
  </span>
</div>
`
